geth/utils/validation.py

```diff
@@ -97,3 +94,69 @@ def validate_geth_kwargs(geth_kwargs: dict[str, Any]) -> bool:
         # TODO more specific error message
         raise ValueError("Invalid geth_kwargs")
     return True
+
+
+class GenesisDataTypedDict(TypedDict, total=False):
+    alloc: dict[str, dict[str, Any]]
+    coinbase: str
+    config: dict[str, Any]
+    difficulty: str
+    extraData: str
+    gasLimit: str
+    mixhash: str
+    nonce: str
+    parentHash: str
+    timestamp: str
+
+
+class GenesisData(BaseModel):
+    alloc: dict[str, dict[str, Any]] = {}
+    coinbase: str = "0x3333333333333333333333333333333333333333"
+    config: dict[str, Any] = {
+        "ethash": {},
+        "homesteadBlock": 0,
+        "daoForkBlock": 0,
+        "daoForkSupport": True,
+        "eip150Block": 0,
+        "eip155Block": 0,
+        "eip158Block": 0,
+        "byzantiumBlock": 0,
+        "constantinopleBlock": 0,
+        "petersburgBlock": 0,
+        "istanbulBlock": 0,
+        "berlinBlock": 0,
+        "londonBlock": 0,
+        "arrowGlacierBlock": 0,
+        "grayGlacierBlock": 0,
+        # merge
+        "terminalTotalDifficulty": 0,
+        "terminalTotalDifficultyPassed": True,
+        # post-merge, timestamp is used for network transitions
+        "shanghaiTime": 0,
+        "cancunTime": 0,
+    }
+    difficulty: str = "0x0"
+    extraData: str = (
+        "0x0000000000000000000000000000000000000000000000000000000000000000"
+    )
+    gasLimit: str = "0x47d5cc"
```

Comment:
This is the only field that differs from the `genesis.json` file. Which should we use, or how does one determine a good default?

Reply:
Let's roll with [geth's default](https://github.com/ethereum/go-ethereum/blob/master/params/protocol_params.go#L29) for both? I'm hesitant to use `0` [and let geth fill it in](https://github.com/ethereum/go-ethereum/blob/master/core/genesis.go#L443-L445), in case they change the way they set their default and this breaks in the future, but that could also be an option if we decide it's OK.

Reply:
updated to geth default 4712388 -> 0x47e7c4